src/swap/SwapScreen.tsx

```diff
@@ -114,6 +113,46 @@ export function SwapScreenSection({ showDrawerTopNav }: { showDrawerTopNav: bool
     [swapAmount]
   )
 
+  const calculatePriceImpactData = () => {
+    if (!exchangeRate) {
+      return { isPriceImpactMissing: false, priceImpactExceedsThreshold: false }
+    }
+
+    const priceImpact = exchangeRate.estimatedPriceImpact
+      ? new BigNumber(exchangeRate.estimatedPriceImpact).dividedBy(100)
+      : null
+
+    const isPriceImpactMissing = priceImpact === null
+    const priceImpactExceedsThreshold =
+      !!priceImpact && priceImpact.gte(priceImpactWarningThreshold)
+
+    return { isPriceImpactMissing, priceImpactExceedsThreshold, priceImpact }
+  }
+
+  const { isPriceImpactMissing, priceImpactExceedsThreshold } = useMemo(() => {
+    return calculatePriceImpactData()
+  }, [exchangeRate])
+
+  useEffect(() => {
+    if (!exchangeRate) {
+      return
+    }
+
+    const { isPriceImpactMissing, priceImpactExceedsThreshold, priceImpact } =
+      calculatePriceImpactData()
+
+    if (priceImpactExceedsThreshold || isPriceImpactMissing) {
+      ValoraAnalytics.track(SwapEvents.swap_price_impact_warning_displayed, {
+        toToken: exchangeRate.toTokenAddress,
+        fromToken: exchangeRate.fromTokenAddress,
+        amount: parsedSwapAmount[updatedField].toString(),
+        amountType: updatedField === Field.FROM ? 'sellAmount' : 'buyAmount',
+        priceImpact: priceImpact?.toString(),
+        provider: exchangeRate.provider,
+      })
+    }
+  }, [exchangeRate])
```

Comment:
nit: there is already a `useEffect` that is triggered on `exchangeRate` change, can we put this logic there?